backend-lib/src/main/rust/utils.rs

```diff
@@ -51,26 +52,42 @@ pub(crate) fn rust_bytes_to_java<'a>(env: &JNIEnv<'a>, data: &[u8]) -> JNIResult
     Ok(jret)
 }
 
-pub(crate) fn rust_vec_to_java<'a, T, U, V, F, G>(
+pub(crate) fn rust_vec_to_java<'a, T, U, V, F>(
     env: &mut JNIEnv<'a>,
     data: Vec<T>,
```

Comment:
This doesn't actually need to be a `Vec` and could be a slice, which gives more flexibility to the caller. Should I change that as well?